app/src/main/java/openfoodfacts/github/scrachx/openfood/models/Ingredient.java

```diff
@@ -25,21 +26,55 @@
     @Unique
     private String tag;
 
+    @Unique
+    private String wikiDataId;
+
+    private Boolean isWikiDataIdPresent;
+
     @ToMany(joinProperties = {
         @JoinProperty(name = "tag", referencedName = "ingredientTag")
     })
     private List<IngredientName> names;
+
+    @ToMany(joinProperties = {
+        @JoinProperty(name = "tag", referencedName = "parentTag")
+    })
+    private List<IngredientsRelation> parents;
+
+    @ToMany(joinProperties = {
+        @JoinProperty(name = "tag", referencedName = "childTag")
+    })
+    private List<IngredientsRelation> children;
 /** Used to resolve relations */
 @Generated(hash = 2040040024)
 private transient DaoSession daoSession;
 /** Used for active entity operations. */
 @Generated(hash = 942581853)
 private transient IngredientDao myDao;
 
-@Generated(hash = 719810036)
-public Ingredient(Long id, String tag) {
+@Generated(hash = 357329675)
+public Ingredient(Long id, String tag, String wikiDataId, Boolean isWikiDataIdPresent) {
     this.id = id;
     this.tag = tag;
+    this.wikiDataId = wikiDataId;
+    this.isWikiDataIdPresent = isWikiDataIdPresent;
+}
+@Keep
+public Ingredient(String tag, List<IngredientName> names, List<IngredientsRelation> parents, List<IngredientsRelation> children, String wikiDataId) {
+    this.tag = tag;
+    this.names = names;
+    this.parents = parents;
+    this.children = children;
+    this.wikiDataId = wikiDataId;
+    this.isWikiDataIdPresent = true;
```

Comment:
Isn't **isWikiDataIdPresent** redundant since we can check by null/empty the **wikiDataId** itself?

Reply:
Yes it is like in Allergen.java ;-)

Reply:
I'll remove it in #2059